components/chapters/LoginModal.tsx

```diff
@@ -0,0 +1,196 @@
+        </button>
+      </div>
+
+      <div className="p-4">
+      {!loaded && !props.signedIn && (
+        <>
+          <img src="/assets/avatars/1.png" width={80} height={80} />
+          <h2 className="mb-4 mt-5 text-3xl font-bold">Load your progress</h2>
+          <p className="mb-5 text-xl">
+            Enter your personal code below to restore your chapter and challenge progress, as well as your avatar. <a href="#" className='text-white underline'>No code yet?</a>
+          </p>
+
+          <div className="flex flex-col justify-between">
+            <div className="flex w-full">
+              <input
+                className="bg-transparent outline-none text-white border-dotted border-white border-2 p-1 w-full text-lg"
+                type="text"
+                placeholder="Enter your code"
+                value={userPrivateKey}
+                onChange={(e) => setPrivateKey(e.target.value)}
+              />
+            </div>
+            <div className="flex w-full mt-4">
+              <button disabled={!userPrivateKey} onClick={loadProgress} className={`border-white border-2 w-full p-1 text-xl ${!userPrivateKey && 'opacity-50'}`}>Load my progress</button>
+            </div>
+            <div className="flex w-full mt-4 mb-4">
+              {userPrivateKey && !checkValidKey() && (
+                <div className='flex items-center'>
+                  <svg width="30" height="30" viewBox="0 0 30 30" fill="none" xmlns="http://www.w3.org/2000/svg">
+                    <rect width="30" height="30" rx="15" fill="#F3241D" fill-opacity="0.25"/>
+                    <circle cx="15" cy="10.25" r="1.25" fill="#F3241D"/>
+                    <rect x="14" y="13" width="2" height="8" fill="#F3241D"/>
+                  </svg>
+                  <p className="text-red-500 text-sm ml-2">Sorry, that’s not the right code for the progress stored in this browser.</p>
+                </div>
+              )}
+            </div>
+
+            <div className='flex items-center mt-auto'>
+                <p>
+                  Want to start again? <button onClick={clearProgress} className='underline'>Clear saved progress</button>.
+                </p>
+            </div>
+          </div>
+        </>)}
+      {props.signedIn && (
+        <>
+          <Image
+            alt="avatar"
+            src={`/assets/avatars/${props.user.avatar}.png`}
+            width={80}
+            height={80}
+            className="rounded-full mb-2"
+          />
+          <h2 className="mb-2 text-3xl font-bold">Load your progress</h2>
+          <p className="mb-5 text-xl">
+            Your progress is stored in this browser. Use the code below to restore it after taking a break.
+          </p>
+
+        <pre className="mb-5 flex flex-col rounded-md border-2 border-dotted border-white/25 p-4">
+          <code className="whitespace-pre-wrap break-all">{props.user.privateKey}</code>
+          {copied ? (
+            <button
```

Comment:
This can be a BoxButton. May need to add a "success" or "clicked" state though.

Reply:
I have difficulties making it work with the hover class and can't override it easily.
I think a follow-up PR should improve this class to allow for other background colors and other hover states.

Reply:
100%. I created issue #50 for this.